### PR TITLE
feat: added applicationPage initial screen

### LINKED ISSUE #54: Application page : Initial screen
**Story**
As a user when I navigate to the application form I should see two tabs ( mentor and mentee) with empty content.
[design link](https://www.figma.com/file/5KpDnPPmrZhudGESduPhq3/M3ntorship-Website?node-id=1089%3A8106)

**functionality**
- A user can switch between the mentor and mentee form by clicking on the associated tab
- The selected tab style should be changed as per the design 
- The progress bar should represent the current step 
- Click on the next button and go to the next step
- The previous button will appear after step 2 and clicking on it will navigate to the previous step 
- The title of the form will change after a user select the type 
- Click on the cancel button should show the UI of the first image attached below 👇 

**Acceptance criteria**

Screen Shot 2022-08-21 at 10 26 01 PM
Screen Shot 2022-08-21 at 10 26 23 PM
Screen Shot 2022-08-21 at 11 09 16 PM

**Note**

we will add form qouestions  in a separate task

### PR DESCRIPTION
**Pull request by AhmedAdel999**
resolves #54.
Note: `ApplicationsTab` might need some refactoring.

### CODE REVIEW

**Review by atwfk — COMMENTED** [review]
move all assets to public folder -> images -> application-page

File: src/modules/ApplicationsPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { ReactElement, useState } from 'react';
+
+import ApplicationsTab from '@/modules/ApplicationsPage/components/ApplicaitonsTab';
+import ApplicationsHeader from '@/modules/ApplicationsPage/components/ApplicationsHeader';
+import MenteeApplication from '@/modules/ApplicationsPage/components/MenteeApplication';
+import MentorApplication from '@/modules/ApplicationsPage/components/MentorApplication';
+
+export default function ApplicationsPage(): ReactElement {
+  const [activeTab, setActiveTab] = useState('mentee');
+  const [currentStep, setCurrentStep] = useState(3);
```
Comment (elhaw) [inline src/modules/ApplicationsPage/index.tsx]: let us add types here

**Review by elhaw — COMMENTED** [review]
My comment is just regarding the folder structure pattern for each page. The components folder in each module will have subfolders( atoms, molecules, organisms) as we did on the home page. Inside the atoms folders, we create an `index` file in which we export all components inside

**Comment by AhmedAdel999** [comment]
I have edited the folder structure in #56.

File: src/modules/ApplicationsPage/components/ApplicaitonsTab/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import clsx from 'clsx';
+import { ButtonHTMLAttributes, ReactElement, ReactNode } from 'react';
+
+interface ApplicaitonsTabProps extends ButtonHTMLAttributes<HTMLButtonElement> {
```
Comment (atwfk) [inline src/modules/ApplicationsPage/components/ApplicaitonsTab/index.tsx]: Please correct the spelling use of `ApplicationsTabProps`

File: src/modules/ApplicationsPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import ApplicationsTab from '@/modules/ApplicationsPage/components/ApplicaitonsTab';
+import ApplicationsHeader from '@/modules/ApplicationsPage/components/ApplicationsHeader';
+import MenteeApplication from '@/modules/ApplicationsPage/components/MenteeApplication';
+import MentorApplication from '@/modules/ApplicationsPage/components/MentorApplication';
+
+export default function ApplicationsPage(): ReactElement {
+  const [activeTab, setActiveTab] = useState<string>('mentee');
+  const [currentStep, setCurrentStep] = useState<number>(3);
+
+  const menteeHanleClick = (): void => {
```
Comment (atwfk) [inline src/modules/ApplicationsPage/index.tsx]: Please correct the spelling use of `menteeHandleClick`

File: src/modules/ApplicationsPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    setActiveTab('mentee');
+  };
+
+  const mentorHanleClick = (): void => {
```
Comment (atwfk) [inline src/modules/ApplicationsPage/index.tsx]: Please correct the spelling use of `mentorHandleClick`

File: src/modules/ApplicationsPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { Typography } from 'm3ntorship-ui';
+import Mentee from 'public/images/applications-page/mentee.svg';
+import Mentor from 'public/images/applications-page/mentor.svg';
+import { ReactElement, useState } from 'react';
+
+import ApplicationsTab from '@/modules/ApplicationsPage/components/ApplicaitonsTab';
```
Comment (atwfk) [inline src/modules/ApplicationsPage/index.tsx]: Please correct the spelling use of `ApplicationsTab` folder name

File: src/modules/ApplicationsPage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import ApplicationsHeader from '@/modules/ApplicationsPage/components/ApplicationsHeader';
+import MenteeApplication from '@/modules/ApplicationsPage/components/MenteeApplication';
+import MentorApplication from '@/modules/ApplicationsPage/components/MentorApplication';
+
+export default function ApplicationsPage(): ReactElement {
+  const [activeTab, setActiveTab] = useState<string>('mentee');
```
Comment (atwfk) [inline src/modules/ApplicationsPage/index.tsx]: the first view in this page on [Figma](https://www.figma.com/file/5KpDnPPmrZhudGESduPhq3/M3ntorship-Website?node-id=1089%3A8106) only contains the tabs, so this should be null in the initial render
```suggestion
  const [activeTab, setActiveTab] = useState<null | 'mentee' | 'mentor'>(null);
```
and would be better to create an enum for them

**Review by atwfk — APPROVED** [review]
LGTM

**Pull request merged**